File: app/src/main/java/com/bus/routes/module/ApplicationModule.java
package com.bus.routes.module;

import android.app.Application;
import android.content.Context;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Gustavo on 23/06/18.
 */

@Module
public class ApplicationModule {

    private final Application mApplication;

    public ApplicationModule(Application app) {
        mApplication = app;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }

}

File: app/src/main/java/com/bus/routes/module/ApplicationModule.kt
package com.bus.routes.module

import android.app.Application
import android.content.Context

import dagger.Module
import dagger.Provides

/**
 * Created by Gustavo on 23/06/18.
 */

@Module
class ApplicationModule(private val mApplication: Application) {

    @Provides
    @ApplicationContext
    internal fun provideContext(): Context {
        return mApplication
    }

    @Provides
    internal fun provideApplication(): Application {
        return mApplication
    }

}
